examples/magento-graphcms/pages/oci.tsx

```diff
@@ -0,0 +1,121 @@
+// TODO: Remove this page once OCI integration is complete.
+import { ApolloErrorAlert, WaitForQueries } from '@graphcommerce/ecommerce-ui'
+import { PageOptions } from '@graphcommerce/framer-next-pages'
+import { cacheFirst } from '@graphcommerce/graphql'
+import { PageMeta, StoreConfigDocument } from '@graphcommerce/magento-store'
+import { FullPageMessage, GetStaticProps, icon404, IconSvg } from '@graphcommerce/next-ui'
+import { i18n } from '@lingui/core'
+import { Trans } from '@lingui/react'
+import { Box, CircularProgress, Container } from '@mui/material'
+import { useOciCart } from '@graphcommerce/magento-oci'
+import { useRouter } from 'next/router'
+import {
+  LayoutDocument,
+  LayoutMinimalProps,
+  LayoutNavigation,
+  LayoutNavigationProps,
+} from '../components'
+import { graphqlSharedClient, graphqlSsrClient } from '../lib/graphql/graphqlSsrClient'
+
+type Props = Record<string, unknown>
+type GetPageStaticProps = GetStaticProps<LayoutMinimalProps, Props>
+
+function CheckoutProcessToken() {
+  const router = useRouter()
+  const { username, password, HOOK_URL: hookUrl } = router.query
+  const { loading, error } = useOciCart()
+
+  const missingParam = router.isReady && (!username || !password || !hookUrl)
+  const isLoading = loading || router.isReady
+  const hasErrors = missingParam || !!error
```

Comment:
I would move all this logic to a separate hook `useOciStartSession` or something?